src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -240,7 +240,7 @@ protected function isValidFile($file)
      *
      * @param  string|null  $key
      * @param  mixed  $default
-     * @return \Illuminate\Http\UploadedFile|\Illuminate\Http\UploadedFile[]|array|null
+     * @return ($key is null ? array<int, \Illuminate\Http\UploadedFile|array<int, \Illuminate\Http\UploadedFile>> : \Illuminate\Http\UploadedFile|\Illuminate\Http\UploadedFile[]|null)
```

Comment:
when `$key` is `null`, the return type is always an array of files, otherwise it can be either an instance of a file, an array of files, or `null`